main.js

```diff
@@ -149,12 +149,14 @@ module.exports = function (argumentOptions) {
         connectionOptions.tls = options.tls;
     }
 
-    if (options.routes) {
-        connectionOptions.routes = options.routes;
+    if (options.connections.routes) {
+        console.log("----- received and setting a route config from calling application");
+        server.settings.connections.routes.files = options.connections.routes.files;
     }
 
-    if (options.state) {
-        connectionOptions.state = options.state;
+    if (options.connections.routes.state) {
+        console.log("----- received and setting a state config from calling application");
+        server.settings.connections.routes.state = options.connections.routes.state;
     }
```

Comment:
Please remove the console.log statements. These proposed changes create errors when running the example app:

```
node example/app.js
```

Error

```
TypeError: Cannot read property 'routes' of undefined
    at module.exports (/Users/adslaton/dev/we/cnn-hapi/main.js:152:28)
    at Object.<anonymous> (/Users/adslaton/dev/we/cnn-hapi/example/app.js:20:24)
    at Module._compile (module.js:556:32)
    at Object.Module._extensions..js (module.js:565:10)
    at Module.load (module.js:473:32)
    at tryModuleLoad (module.js:432:12)
    at Function.Module._load (module.js:424:3)
    at Module.runMain (module.js:590:10)
    at run (bootstrap_node.js:394:7)
    at startup (bootstrap_node.js:149:9)
```

You will need to check for the existence of options.connections, options.connections.routes, and options.connections.routes.state before trying to assign the value. 

Your proposed changes present a need to change how options are being passed into cnn-hapi. We should allow all the core hapi options to be available. I will work on revising how these options are being passed and set.